components/patient-data/ui/src/main/resources/PhenoTips/PatientSheetCode.xml

```diff
@@ -2005,6 +2005,23 @@ document.observe('xwiki:dom:loaded', function() {
       new PhenoTips.widgets.ChartGenerator($('charts'), chartDataSources);
     });
 
+    //create a warning when measurements are being entered buyt no birth date is specified
+    var dateOfBirth=$(document.documentElement).down('input[name$=date_of_birth]');
+    var previousElement= $$('.measurement-info &gt; .expand-tools')[0] ? $$('.measurement-info &gt; .expand-tools')[0] : $('HMeasurements');
+    if (dateOfBirth &amp;&amp; previousElement){
+        _updateNoBirthDateWarning();
+        dateOfBirth.observe('blur', _updateNoBirthDateWarning);
+    }
+    function _updateNoBirthDateWarning(){
+      birthdate = dateOfBirth.value;
+      if (!birthdate &amp;&amp; !$('no-birth-date-warning')){
+        previousElement.insert({
```

Comment:
Now the issue is that the value of `previousElement` doesn't change, but the needed insertion point might change since `expand-tools` will probably show up. What you should do:

- if the _birthdate_ field is present, and the _Measurements_ section is present, then register the listener
- when the event is triggered, find the insertion point and insert the warning after it